src/table/BaseColumnView.cc

```diff
@@ -69,6 +69,11 @@ std::shared_ptr<BaseTable> BaseColumnView::getTable() const { return _impl->tabl
 
 template <typename T>
 typename ndarray::ArrayRef<T, 1> const BaseColumnView::operator[](Key<T> const &key) const {
+    if (!key.isValid()) {
+        throw LSST_EXCEPT(
+            pex::exceptions::LogicError,
+            "Key is not valid (if this is a SourceCatalog, make sure slot aliases have been set up).");
```

Comment:
Why `SourceCatalog` here? Can slots not be set on `SimpleCatalog`s? What other types of catalogs can or cannot have slots?